workflow/scripts/s09_cell_to_xarray.py

```diff
@@ -0,0 +1,236 @@
+    synced_timestamp_files = synced_timestamp_files, 
+    spike_clusters_files = spike_clusters_files,
+    trial_times = trial_times, 
+    trial_window = trial_window,
+    bin_duration = bin_duration)
+
+
+# %% Applying Kernel to binned firing rate
+# if we want the gaussian to have sigma = 0.2s
+time_for_1SD = 0.2 # sec
+sigma = time_for_1SD * (1000/bin_duration)
+
+# May not be applied until bins are aggregated by trials
+# Return the binned spike array convoluted by half-gaussian window (so FR do not increases before spike [but binning])
+convoluted_binned_array = halfgaussian_filter1d(all_probes_binned_array_averaged, sigma = sigma, axis=-1, output=None,
+                      mode="constant", cval=0.0, truncate=4.0) # truncate the window at 4SD
+
+scaler = StandardScaler()
+# z-scored firing rate
+# .T).T to z-score along time dimension
+z_conv_FR = scaler.fit_transform(convoluted_binned_array.T).T
+
+# %% Combining spikes rates and scored
+
+spike_fr_xr = xr.DataArray(
+    convoluted_binned_array,
+    name = 'spikes_FR',
+    coords={'UID': all_clusters_UIDs, 'time':spike_time_bins[1:]},
+    dims=('UID', 'time')
+)
+
+spike_zscored_xr = xr.DataArray(
+    z_conv_FR,
+    name = 'spikes_Zscore',
+    coords={'UID': all_clusters_UIDs, 'time':spike_time_bins[1:]},
+    dims=('UID', 'time')
+)
+
+# %% Combine cell metrics and spike bins
+
+session_cell_metrics = pd.DataFrame(data={'UID':all_clusters_UIDs})
+session_cell_metrics.set_index('UID', inplace=True)
+uids = list()
+for f_idx, cell_metrics_file in enumerate(cell_metrics_files):
+    cell_metrics_df = pd.read_pickle(cell_metrics_file)
+    session_cell_metrics = pd.concat([session_cell_metrics,cell_metrics_df])
+    uids = uids + cell_metrics_df.index.tolist()
+
+
+# add clusters_UIDs from spike_clusters.npy + those of cell metrics and merge
+uids = list(set(uids+all_clusters_UIDs))
+
+cluster_cell_IDs = pd.DataFrame(data={'UID':all_clusters_UIDs})
+# Add sorted UIDs without cell metrics :  To investigate maybe some units not only present before / after 1st rsync?
+session_cell_metrics = cell_metrics_df.merge(cluster_cell_IDs, on='UID', how='outer',)
+
+session_cell_metrics.set_index('UID', inplace=True)
+
+# A bit of tidy up is needed after merging so str columns can be str and not objects due to merge
+session_cell_metrics = dataframe_cleanup(session_cell_metrics)
+
+xr_cell_metrics = session_cell_metrics.to_xarray()
+
+# Create the xr dataset
+xr_spikes_averaged = xr.merge([spike_fr_xr, spike_zscored_xr, xr_cell_metrics])
+xr_spikes_averaged.attrs['bin_duration'] = bin_duration
+
+# Save
+xr_spikes_averaged.to_netcdf(Path(sinput.xr_session).parent / 'xr_spikes_averaged.nc', engine='h5netcdf')
+
+
+# %% Preview of cluster responses to trigger
+# Sort clusters by peak time in the trial
+idx_max_FR = np.argmax(convoluted_binned_array,1)
+cluster_ID_sorted = np.argsort(idx_max_FR)
+
+sns.heatmap(spike_zscored_xr[np.flip(cluster_ID_sorted),:], vmin=-3, vmax=3)
```

Comment:
Don't do plotting in a script unless it is saved. Otherwise, it is wasting processing time. No one can see this figure in the terminal